### PR TITLE
ICU-20048 Remove use of std::align() for compat with g++ 4.9.

### PR DESCRIPTION
**Pull request by aheninger**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20048
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/common/ucnv.cpp [hunk 1]
```diff
@@ -225,13 +225,16 @@ ucnv_safeClone(const UConverter* cnv, void *stackBuffer, int32_t *pBufferSize, U
...
     /* Adjust (if necessary) the stackBuffer pointer to be aligned correctly for a UConverter.
+     * TODO(Jira ICU-20736) Redo this using std::align() once g++4.9 compatibility is no longer needed.
      */
-    size_t altStackBufferSize = stackBufferSize;  // Incompatible types for passing by reference.
     if (stackBuffer) {
-        if (std::align(alignof(UConverter), bufferSizeNeeded, stackBuffer, altStackBufferSize)) {
-            stackBufferSize = static_cast<int32_t>(altStackBufferSize);
+        uintptr_t p = reinterpret_cast<uintptr_t>(stackBuffer);
+        uintptr_t aligned_p = (p + alignof(UConverter) - 1) & ~(alignof(UConverter) - 1);
+        ptrdiff_t sizeAdjustment = aligned_p - p;
```
Comment (markusicu) [inline icu4c/source/common/ucnv.cpp]: It's not a sizeAdjustment, it's a pointerAdjustment or offset or alignIncrement... right?

Comment (aheninger) [reply icu4c/source/common/ucnv.cpp]: Well, it's both. It is the amount by which the stack buffer size is about to be adjusted, and the amount by which the pointer was adjusted. I was thinking the former when I named it.

I'm perfectly happy to change the name if you think it's clearer.

Comment (markusicu) [reply icu4c/source/common/ucnv.cpp]: Hm, you are right, but it's confusing looking at the line where it's computed. "offset" would be clearer.

Comment (aheninger) [reply icu4c/source/common/ucnv.cpp]: Done

**Pull request merged**